src/vs/workbench/contrib/scm/common/scm.ts

```diff
@@ -84,6 +85,8 @@ export interface IInputValidator {
 }
 
 export interface ISCMInput {
+	store(value: string): void;
+	storedValue(): string | undefined;
```

Comment:
Thinking ahead (https://github.com/microsoft/vscode/issues/26952)... would a different interface here make more sense? What do you think about having a read-only array of values, acting as a history? Something like:

```ts
export interface ISCMInput {
  readonly history: readonly string[];
}
```

The `ISCMInput` interface should really only expose a read-only value/history. No other entity other than `SCMInput` should be allowed to modify it. So, that history would be updated simply by setting new `value` on `SCMInput`, ie in the setter of `SCMInput.value`.

Reply:
I like that idea and will explore it today.